.github/actions/e2e_test/action.yml

```
@@ -56,6 +56,21 @@ runs:
         echo "Invalid input for test field: ${{ inputs.test }}"
         exit 1
 
+    # Some e2e test may require a special environment.
+    # For those, the environment is checked here prior to running the test.
+    # For example, K-Bench's network benchmarks require at least two distinct worker nodes.
+    - name: Validate e2e test environment
+      if: ${{ contains(fromJson('["k-bench"]'), inputs.test) }}
+      shell: bash
+      run: |
+        if [ "$e2eTest" = "k-bench" ] && [ "$constellationWorkerNodes" -lt 2 ]; then
+          echo "::error::Test K-Bench requires at least 2 worker nodes."
+          exit 1
+        fi
+      env:
+        constellationWorkerNodes: ${{ inputs.workerNodesCount }}
+        e2eTest: ${{ inputs.test}}
```

```suggestion
    - name: Validate k-bench inputs
      if:  inputs.test == 'k-bench'
      shell: bash
      run: |
        if [[ "${{ inputs.workerNodesCount }}" -lt 2 ]]; then
          echo "::error::Test K-Bench requires at least 2 worker nodes."
          exit 1
        fi
```

Would suggest to not over-complicate for now. It might be easier to just add a step for each test that requires it instead of building huge bash constructs.